BuildError
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'add_table_widget'. Did you mean 'add_app_user' instead?

Traceback (most recent call last)
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\app.py", line 2213, in __call__
return self.wsgi_app(environ, start_response)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\app.py", line 2193, in wsgi_app
response = self.handle_exception(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\app.py", line 2190, in wsgi_app
response = self.full_dispatch_request()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\app.py", line 1486, in full_dispatch_request
rv = self.handle_user_exception(e)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\app.py", line 1484, in full_dispatch_request
rv = self.dispatch_request()
     ^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\app.py", line 1469, in dispatch_request
return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
return current_app.ensure_sync(func)(*args, **kwargs)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "d:\VScode\VScode\flaskwebapp\app1.py", line 678, in footprint_database
return render_template('footprint.html',
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\templating.py", line 151, in render_template
return _render(app, template, context)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\templating.py", line 132, in _render
rv = template.render(context)
     ^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\jinja2\environment.py", line 1295, in render
self.environment.handle_exception()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\jinja2\environment.py", line 942, in handle_exception
raise rewrite_traceback_stack(source=source)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "d:\VScode\VScode\flaskwebapp\templates\footprint.html", line 1, in top-level template code
{% extends "base.html" %}
File "d:\VScode\VScode\flaskwebapp\templates\base.html", line 264, in top-level template code
{% block content %}{% endblock %}
File "d:\VScode\VScode\flaskwebapp\templates\footprint.html", line 176, in block 'content'
<form method="POST" action="{{ url_for('add_table_widget') }}">
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\app.py", line 1697, in url_for
return self.handle_url_build_error(error, endpoint, values)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\flask\app.py", line 1686, in url_for
rv = url_adapter.build(  # type: ignore[union-attr]
     
File "D:\VScode\VScode\flaskwebapp\.venv\Lib\site-packages\werkzeug\routing\map.py", line 950, in build
raise BuildError(endpoint, values, method, self)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'add_table_widget'. Did you mean 'add_app_user' instead?
The debugger caught an exception in your WSGI application. You can now look at the traceback which led to the error.
To switch between the interactive traceback and the plaintext one, you can click on the "Traceback" headline. From the text traceback you can also create a paste of it. For code execution mouse-over the frame you want to debug and click on the console icon on the right side.

You can execute arbitrary Python code in the stack frames and there are some extra helpers available for introspection:

dump() shows all variables in the frame
dump(obj) dumps all that's known about the object
Brought to you by DON'T PANIC, your friendly Werkzeug powered traceback interpreter.
